codegen/masm/src/masm/program.rs

```diff
@@ -499,8 +516,54 @@ impl Library {
             .iter()
             .map(|rodata| (rodata.digest, rodata.to_elements()))
             .collect();
-        Ok(Arc::new(lib.with_advice_map(advice_map)))
+        let mut mast_forest = lib.mast_forest().as_ref().clone();
+        mast_forest.advice_map_mut().extend(advice_map);
+        let converted_exports = recover_wasm_cm_interfaces(&lib);
+        let lib = CompiledLibrary::new(Arc::new(mast_forest), converted_exports)?;
```

Comment:
Note: When wiring up HIR2 to the frontend, the lowering from Wasm to HIR will need to create `Interface` operations in the top-level `Component` which contain `Function` ops corresponding to the exports of component instances in the original Wasm component. The `Component` operation will also contain component-level function exports and `Module`s corresponding to the core Wasm modules present in the original Wasm component. See [Component Translation](https://github.com/0xPolygonMiden/compiler/discussions/360) for further details.

The new codegen crate no longer has its own Miden Assembly IR, so the `Library` and `Program` structs no longer exist. To the extent that some of this stuff still exists (by which I mean things like assembling to MAST, late-stage linking and the like), it will either be done in the _new_ `Linker` (which is much simpler than the current one, and runs just prior to codegen, rather than as a initial part of lowering into HIR), or in the compiler driver (i.e. in an appropriate `Stage` of the compiler). I expect that a lot of this will no longer be necessary though, since the new representation is more faithful to the Component Model, so fewer contortions are required to lower into Miden Assembly than we currently are forced to do.